restate-test-utils/src/test_utils.rs

```diff
@@ -0,0 +1,211 @@
+use nu_ansi_term::Style;
+use reqwest::Response;
+use testcontainers::{core::{IntoContainerPort, WaitFor}, runners::AsyncRunner, ContainerAsync, ContainerRequest, GenericImage, ImageExt};
+use serde::{Serialize, Deserialize};
+use restate_sdk::{discovery::Service, errors::HandlerError, prelude::{Endpoint, HttpServer}};
+use tokio::{io::{self, AsyncWriteExt}, task::{self, JoinHandle}};
+use std::time::Duration;
+
+// addapted from from restate-admin-rest-model crate version 1.1.6
+#[derive(Serialize, Deserialize, Debug)]
+pub struct RegisterDeploymentRequestHttp {
+    uri: String,
+    additional_headers:Option<Vec<(String, String)>>,
+    use_http_11: bool,
+    force: bool,
+    dry_run: bool
+}
+
+#[derive(Serialize, Deserialize, Debug)]
+pub struct RegisterDeploymentRequestLambda {
+    arn: String,
+    assume_role_arn: Option<String>,
+    force: bool,
+    dry_run: bool,
+}
+
+#[derive(Serialize, Deserialize, Debug)]
+struct VersionResponse {
+    version:String,
+    min_admin_api_version:u32,
+    max_admin_api_version:u32
+}
+
+pub struct TestContainer {
+    container:ContainerAsync<GenericImage>,
+    stdout_logging:JoinHandle<()>,
+    stderr_logging:JoinHandle<()>,
+    endpoint:Option<JoinHandle<()>>
+}
+
+
+impl TestContainer {
+
+    //"docker.io/restatedev/restate", "latest"
+    pub async fn new(image:&str, version:&str) -> Result<Self, HandlerError> {
```

Comment:
You should use `anyhow::Error` here and everywhere else in this file, `HandlerError` is an error type reserved to be used only inside restate service handlers.